src/components/TodoItem.tsx

```diff
@@ -0,0 +1,110 @@
+import { ChangeEvent, FC, KeyboardEvent, useState } from 'react';
+import { Todo } from '../types';
+import { useTodos } from '../utils/TodosContext';
+import cn from 'classnames';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoItem: FC<Props> = ({ todo }) => {
+  const { removeTodo, loadingTodosIds, toggleOne, updateTitle } = useTodos();
+  const [idToEdit, setIdToEdit] = useState<number | null>(null);
+  const [textToEdit, setTextToEdit] = useState(todo.title);
+  const isLoadingActive = loadingTodosIds.includes(todo.id) || todo.id === 0;
+
+  const toggleEditTodo = (todoId: number) => {
+    setIdToEdit(todoId);
+    setTextToEdit(todo.title);
+  };
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setTextToEdit(e.target.value);
+  };
+
+  const handleRename = async () => {
+    const newText = textToEdit.trim();
+
+    if (newText === todo.title) {
+      setIdToEdit(null);
+
+      return;
+    }
+
+    if (!newText) {
+      removeTodo(todo.id);
+    } else {
+      updateTitle(todo.id, newText, async () => setIdToEdit(null));
```

Comment:
```suggestion
      updateTitle(todo.id, newText, () => setIdToEdit(null));
```

for what we need here async callback?